main.go

```diff
@@ -1060,17 +1063,10 @@ func transformGo(file *ast.File, info *types.Info, blacklist map[types.Object]st
 		// log.Printf("%#v %T", node, obj)
 		switch x := obj.(type) {
 		case *types.Var:
-			if x.IsField() && x.Exported() {
-				// might be used for reflection, e.g.
-				// encoding/json without struct tags
-				return true
-			}
-
-			if obj.Parent() != pkg.Scope() {
+			if obj.Parent() != nil && obj.Parent() != pkg.Scope() {
```

Comment:
simplify: `if parent := obj.Parent(); parent != nil && parent != pkg.Scope() {`

Reply:
to clarify, this is not about making the code shorter (it isn't), but about deduplicating method calls. it's also more obvious that both checks refer to the same value this way.

Reply:
Yeah that makes complete sense, should have caught this myself